### ISSUE TITLE
ApplicationInsights:Sender [ { Error: read ECONNRESET

### ISSUE THREAD

**Issue by tirish** [Microsoft/ApplicationInsights-node.js#378]
```
ApplicationInsights:Sender [ { Error: read ECONNRESET 
  at TLSWrap.onread (net.js:568:26) code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' } ]
```

We are currently experiencing this in our Node apps (started around 2:55 pm ET, still ongoing as of 3:45 pm ET). This happened Friday and we believe was related to App Insights Downtime: https://blogs.msdn.microsoft.com/applicationinsights-status/

Is this definitely the result of downtime? If so, can the error be improved? If not, any idea what is going on here?

We are using version 1.0.1

**Comment by OsvaldoRosado** [comment]
In general with this error, all that's for certain is that the client is unable to reach the backend. While it could certainly be a problem with the ingestion service, I'm not aware of any incidents with the service itself that have persisted from Friday to now.

In the default configuration the SDK will store failed telemetry to disk and retry later when the connection is healthy, so you can take these errors as informational as to the state of the connection to our backend.

The error could be made "prettier" but I'm not sure if that's the best call here. From the client side there's no way to know the true cause of the connectivity error, so repeating the error that Node actually gives is the most informational we can be.

**Comment by OutKa5t** [comment]
@OsvaldoRosado I don't think this issue relates to AppInsights status because I see it daily with Node Apps on App Services but most telemetry makes it through. Can you tell me whether the stderr is written when the telemetry is stored to disk or when the retry occurs? 

If it really is informational then should it not be recategorised? How can I tell if the retry has worked or even what the telemetry is that failed to send?

How can I investigate this further?

**Comment by OsvaldoRosado** [comment]
This message is now recategorized in the just-released version 1.0.3.

If retries are enabled connection errors are only reported if verbose logging is enabled, unless 5 consecutive connection errors occur. Every 5 consecutive errors, an error-level warning (reported by default) is triggered to disclose the potential data loss from continued inability to reach the backend.

With this change the answer to:
> How can I tell if the retry has worked or even what the telemetry is that failed to send?
Becomes: You can tell because you now won't get errors if things are working okay :smile:

**Comment by OsvaldoRosado** [comment]
Closing this for now. Feel free to reopen if this is still an issue!
